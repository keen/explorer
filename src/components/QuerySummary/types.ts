import { Timeframe, Timezones } from '@keen.io/query-creator';

export type Filter = {
  operator: string;
  property_name: string;
  property_value: string;
  property_type?: string;
};

export type FunnelStep = {
  actor_property: string;
  event_collection: string;
  inverted: boolean;
  optional: boolean;
  timeframe: Timeframe;
  timezone?: Timezones | number;
  with_actors: boolean;
  filters: Filter[];
};
