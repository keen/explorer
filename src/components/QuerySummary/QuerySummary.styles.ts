import styled from 'styled-components';

export const Wrapper = styled.div`
  padding: 20px;
`;
