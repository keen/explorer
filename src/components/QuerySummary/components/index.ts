import Timeframe from './Timeframe';
import FunnelSteps from './FunnelSteps';
import PropertyName from './PropertyName';
import StyledTable from './Table';

export { Timeframe, FunnelSteps, PropertyName, StyledTable };
