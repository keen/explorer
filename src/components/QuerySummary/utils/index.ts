import { transformName } from './transformName';

export { transformName };
