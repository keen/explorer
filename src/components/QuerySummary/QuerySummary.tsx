import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';

import {
  Timeframe,
  FunnelSteps,
  PropertyName,
  StyledTable,
} from './components';
import { Wrapper } from './QuerySummary.styles';

type Props = {
  /** Query settings */
  querySettings: Record<string, any>;
  chartSettings: Record<string, any>;
};

const QuerySummary: FC<Props> = ({ querySettings, chartSettings }) => {
  const { t } = useTranslation();
  const {
    query: {
      analysis_type: analysisType,
      event_collection: eventCollection,
      target_property: targetProperty,
      timeframe,
      timezone,
      filters,
      steps,
    },
  } = querySettings;

  const { stepLabels } = chartSettings;

  return (
    <Wrapper>
      <StyledTable.Table>
        <StyledTable.Body>
          {analysisType && (
            <StyledTable.Row>
              <StyledTable.Label>
                {t('query_summary.analysis')}
              </StyledTable.Label>
              <StyledTable.Value>{analysisType}</StyledTable.Value>
            </StyledTable.Row>
          )}
          {eventCollection && (
            <StyledTable.Row>
              <StyledTable.Label>
                {t('query_summary.event_stream')}
              </StyledTable.Label>
              <StyledTable.Value>{eventCollection}</StyledTable.Value>
            </StyledTable.Row>
          )}
          {targetProperty && (
            <StyledTable.Row>
              <StyledTable.Label>
                {t('query_summary.target_property')}
              </StyledTable.Label>
              <StyledTable.Value>
                <PropertyName name={targetProperty} />
              </StyledTable.Value>
            </StyledTable.Row>
          )}
          {timeframe && (
            <StyledTable.Row>
              <StyledTable.Label>
                {t('query_summary.timeframe')}
              </StyledTable.Label>
              <StyledTable.Value>
                <Timeframe timeframe={timeframe} timezone={timezone} />
              </StyledTable.Value>
            </StyledTable.Row>
          )}
          {!!filters?.length && (
            <StyledTable.Row>
              <StyledTable.Label>
                {t('query_summary.applied_filters')}
              </StyledTable.Label>
              <StyledTable.Value>{filters.length}</StyledTable.Value>
            </StyledTable.Row>
          )}
        </StyledTable.Body>
      </StyledTable.Table>
      {steps && <FunnelSteps steps={steps} stepLabels={stepLabels} />}
    </Wrapper>
  );
};

export default QuerySummary;
