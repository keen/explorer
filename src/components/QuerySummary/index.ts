import QuerySummary from './QuerySummary';

export default QuerySummary;
