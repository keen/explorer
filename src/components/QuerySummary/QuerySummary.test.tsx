/* eslint-disable @typescript-eslint/camelcase */
import React from 'react';
import 'jest-styled-components';
import { render as rtlRender } from '@testing-library/react';

import QuerySummary from './QuerySummary';

const render = (overProps: any = {}) => {
  const props = {
    querySettings: {
      query: {
        analysis_type: 'sum',
        event_collection: 'purchases',
        target_property: 'item.price',
        timezone: 'UTC',
        filters: [
          {
            operator: 'gt',
            property_value: '100',
            property_name: 'item.quantity',
          },
        ],
      },
    },
    chartSettings: {},
    ...overProps,
  };

  const wrapper = rtlRender(<QuerySummary {...props} />);

  return {
    wrapper,
    props,
  };
};

test('should render query summary', () => {
  const {
    wrapper: { container },
  } = render();

  expect(container).toMatchSnapshot();
});

test('should render query summary for funnels', () => {
  const querySettings = {
    query: {
      analysis_type: 'funnel',
      steps: [
        {
          with_actors: false,
          actor_property: 'user.id',
          filters: [],
          timeframe: 'this_14_days',
          timezone: 'UTC',
          event_collection: 'signups',
          optional: false,
          inverted: false,
        },
        {
          with_actors: false,
          actor_property: 'user.id',
          filters: [],
          timeframe: 'this_14_days',
          timezone: 'UTC',
          event_collection: 'purchases',
          optional: false,
          inverted: false,
        },
      ],
    },
  };
  const {
    wrapper: { container },
  } = render({ querySettings });
  expect(container).toMatchSnapshot();
});
