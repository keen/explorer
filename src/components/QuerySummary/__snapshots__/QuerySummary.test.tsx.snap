// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render query summary 1`] = `
.c1 {
  margin: 0 0 10px 0;
  padding: 0;
  border-collapse: collapse;
  table-layout: fixed;
  text-align: left;
}

.c4 {
  padding: 0 20px 10px 0;
  font-family: 'Lato Bold',sans-serif;
  font-size: 14px;
  line-height: 17px;
  color: #4F5B5F;
}

.c6 {
  padding: 0 0 10px 0;
  font-family: 'Lato Regular',sans-serif;
  font-size: 14px;
  line-height: 17px;
  color: #4F5B5F;
}

.c2 {
  vertical-align: top;
}

.c2:last-child .c3,
.c2:last-child .c5 {
  padding-bottom: 0;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  display: inline-block;
  margin-left: 3px;
  margin-right: 3px;
}

.c0 {
  padding: 20px;
}

<div>
  <div
    class="c0"
  >
    <table
      class="c1"
    >
      <tbody
        class=""
      >
        <tr
          class="c2"
        >
          <th
            class="c3 c4"
          >
            query_summary.analysis
          </th>
          <td
            class="c5 c6"
          >
            sum
          </td>
        </tr>
        <tr
          class="c2"
        >
          <th
            class="c3 c4"
          >
            query_summary.event_stream
          </th>
          <td
            class="c5 c6"
          >
            purchases
          </td>
        </tr>
        <tr
          class="c2"
        >
          <th
            class="c3 c4"
          >
            query_summary.target_property
          </th>
          <td
            class="c5 c6"
          >
            <div
              class="c7"
            >
              <span>
                item
              </span>
              <span
                class="c8"
              >
                <svg
                  height="10"
                  opacity="1"
                  viewBox="0 0 30 30"
                  width="10"
                >
                  <polygon
                    fill="#77A3BB"
                    points="9.45351896 0 24.453519 15 9.45351896 30 5.55 26.096481 16.6465765 15 5.55 3.90351896"
                  />
                </svg>
              </span>
            </div>
            <div
              class="c7"
            >
              <span>
                price
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="c2"
        >
          <th
            class="c3 c4"
          >
            query_summary.applied_filters
          </th>
          <td
            class="c5 c6"
          >
            1
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`should render query summary for funnels 1`] = `
.c1 {
  margin: 0 0 10px 0;
  padding: 0;
  border-collapse: collapse;
  table-layout: fixed;
  text-align: left;
}

.c4 {
  padding: 0 20px 10px 0;
  font-family: 'Lato Bold',sans-serif;
  font-size: 14px;
  line-height: 17px;
  color: #4F5B5F;
}

.c6 {
  padding: 0 0 10px 0;
  font-family: 'Lato Regular',sans-serif;
  font-size: 14px;
  line-height: 17px;
  color: #4F5B5F;
}

.c2 {
  vertical-align: top;
}

.c2:last-child .c3,
.c2:last-child .c5 {
  padding-bottom: 0;
}

.c8 {
  border: 1px solid #EDEDED;
}

.c7 + .c7 {
  border-top: none;
}

.c9 {
  padding: 10px 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c10 {
  margin-right: 10px;
}

.c11 {
  margin-right: 10px;
  font-family: 'Lato Bold',sans-serif;
  font-size: 14px;
  line-height: 17px;
  color: #4F5B5F;
}

.c12 {
  font-family: 'Lato Regular',sans-serif;
  font-size: 14px;
  line-height: 17px;
  color: #4F5B5F;
}

.c0 {
  padding: 20px;
}

<div>
  <div
    class="c0"
  >
    <table
      class="c1"
    >
      <tbody
        class=""
      >
        <tr
          class="c2"
        >
          <th
            class="c3 c4"
          >
            query_summary.analysis
          </th>
          <td
            class="c5 c6"
          >
            funnel
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="c7 c8"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <svg
            height="15"
            opacity="1"
            viewBox="0 0 30 30"
            width="15"
          >
            <polygon
              fill="rgba(79,91,95,0.7)"
              points="9.45351896 0 24.453519 15 9.45351896 30 5.55 26.096481 16.6465765 15 5.55 3.90351896"
            />
          </svg>
        </div>
        <div
          class="c11"
        >
          query_summary.step
           
          1
        </div>
        <div
          class="c12"
        >
          signups
        </div>
      </div>
    </div>
    <div
      class="c7 c8"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <svg
            height="15"
            opacity="1"
            viewBox="0 0 30 30"
            width="15"
          >
            <polygon
              fill="rgba(79,91,95,0.7)"
              points="9.45351896 0 24.453519 15 9.45351896 30 5.55 26.096481 16.6465765 15 5.55 3.90351896"
            />
          </svg>
        </div>
        <div
          class="c11"
        >
          query_summary.step
           
          2
        </div>
        <div
          class="c12"
        >
          purchases
        </div>
      </div>
    </div>
  </div>
</div>
`;
